snuba/query/processors/granularity_processor.py

```diff
@@ -39,3 +41,53 @@ def process_query(self, query: Query, query_settings: QuerySettings) -> None:
                 Literal(None, granularity),
             )
         )
+
+
+PERFORMANCE_GRANULARITIES = [(60, 1), (3600, 2), (86400, 3)]
+DEFAULT_MAPPED_GRANULARITY_ENUM = 1
+
+
+class MappedGranularityProcessor(QueryProcessor):
+    """
+    Use the granularity set on the query to filter on the granularity column,
+    supporting generic-metrics style enum mapping
+    """
+
+    def __init__(
+        self,
+        accepted_granularities: Sequence[Tuple[int, int]],
+        default_granularity_enum: int,
+    ):
+        self._accepted_granularities = accepted_granularities
```

Comment:
Maybe I'm being paranoid but I'd probably throw `sorted()` around this just to make sure it's ordered as the next step depends on it. I know it's sorted now but you never know when we might want to add a granularity later and it won't be in order anymore. Could also go ahead and reverse it here so you don't need to do that on every query to be processed.